src/factories.js

```diff
@@ -50,21 +46,19 @@ export function createShorthand(Component, mapValueToProps, val, defaultProps =
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
+    return <Component {...mergePropsAndClassName(mapValueToProps(val), defaultProps)} />
```

Comment:
Indeed I have a branch adding tests currently, will push soon.  This came up from the Input test for `spreads input type` failing since it could no longer override the type.  Again, tests are very much needed and will no doubt solve all this.